In [1]:
import numpy as np
import pandas as pd
!pip install plotly
import plotly.graph_objects as go
import plotly.express as px

In [2]:
df=pd.read_excel("/content/Data set for BA (1).xlsx")

In [3]:
excel_sheets = pd.ExcelFile("/content/Data set for BA (1).xlsx")
sheets = excel_sheets.sheet_names
sheets

['Report Snapshot',
 'User Acquisition',
 'Traffic Aquisition',
 'Event Report',
 'Conversion Report',
 'Pages & Screens Report',
 'Retention Overview',
 'User Engagement Overview',
 'Demographics Report',
 'Citiwise Report',
 'Gender Report',
 'User By Interest',
 'User by Language',
 'User By Age',
 'Google Ads Report']

In [4]:
google_Ads_Report =pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Google Ads Report')
user_by_age = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='User By Age')
user_by_language= pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='User by Language')
user_by_intrest = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name= 'User By Interest')
gender_report = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Gender Report')
citiwise_report = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Citiwise Report')
demographics_report= pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Demographics Report')
pages_and_screens_report = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Pages & Screens Report')
conversion_report = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Conversion Report')
event_report= pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Event Report')
traffic_aquisition = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='Traffic Aquisition')
user_acquisition = pd.read_excel("/content/Data set for BA (1).xlsx",sheet_name='User Acquisition')

**Google Ads Report**

In [6]:
google_Ads_Report.head()

Session Google Ads campaign  Users  Sessions  Engaged sessions  \
0        App Installation for May --Shahid   5429     10936              6276   
1      App Install-States-A200Inst-20Jun22    842      1655               968   
2  App Install-States-B100Installs-22Jun22    742      1332               780   
3          App Install for April -- Shahid    473       976               546   
4  Video-AppInstall-PS-Internships-11Jul22    510       966               515   

   Google Ads clicks  Google Ads cost  Google Ads cost per click  Conversions  \
0             147100       179175.000                   1.218049        12257   
1              28742        24309.130                   0.845770         1794   
2              17809        22374.580                   1.256363         1422   
3              19302        20525.180                   1.063370         1115   
4               9831         6377.833                   0.648747         1032   

   Cost per conversion  Event count  Total revenue  Return on ad spend  
0            14.618180        97802              0                   0  
1            13.550240        15311              0                   0  
2            15.734580        11640              0                   0  
3            18.408230         8001              0                   0  
4             6.180071        10323              0                   0

In [7]:
#checking datatypes
google_Ads_Report.dtypes

Session Google Ads campaign     object
Users                            int64
Sessions                         int64
Engaged sessions                 int64
Google Ads clicks                int64
Google Ads cost                float64
Google Ads cost per click      float64
Conversions                      int64
Cost per conversion            float64
Event count                      int64
Total revenue                    int64
Return on ad spend               int64
dtype: object

In [8]:
#checking missing values
google_Ads_Report.isnull().sum()

Session Google Ads campaign    0
Users                          0
Sessions                       0
Engaged sessions               0
Google Ads clicks              0
Google Ads cost                0
Google Ads cost per click      0
Conversions                    0
Cost per conversion            0
Event count                    0
Total revenue                  0
Return on ad spend             0
dtype: int64

In [9]:
google_Ads_Report.sum()

Session Google Ads campaign    App Installation for May --ShahidApp Install-S...
Users                                                                       9993
Sessions                                                                   19112
Engaged sessions                                                           11350
Google Ads clicks                                                         255318
Google Ads cost                                                     301337.94676
Google Ads cost per click                                              18.566756
Conversions                                                                21323
Cost per conversion                                                   186.155402
Event count                                                               185024
Total revenue                                                                  0
Return on ad spend                                                             0
dtype: object

In [10]:
google_Ads_Report.describe()

Users      Sessions  Engaged sessions  Google Ads clicks  \
count    15.000000     15.000000         15.000000          15.000000   
mean    666.200000   1274.133333        756.666667       17021.200000   
std    1348.253696   2721.749093       1558.314184       36934.069186   
min       2.000000      5.000000          5.000000          14.000000   
25%      62.000000    100.500000         65.500000        1778.500000   
50%     370.000000    610.000000        425.000000        4475.000000   
75%     621.000000    971.000000        654.500000       14202.000000   
max    5429.000000  10936.000000       6276.000000      147100.000000   

       Google Ads cost  Google Ads cost per click   Conversions  \
count        15.000000                  15.000000     15.000000   
mean      20089.196451                   1.237784   1421.533333   
std       44773.537357                   0.635640   3047.587755   
min          16.623960                   0.485200      5.000000   
25%        1399.447500                   0.745567    125.500000   
50%        8839.723000                   1.131950    709.000000   
75%       16304.610000                   1.656639   1073.500000   
max      179175.000000                   2.415885  12257.000000   

       Cost per conversion   Event count  Total revenue  Return on ad spend  
count            15.000000     15.000000           15.0                15.0  
mean             12.410360  12334.933333            0.0                 0.0  
std               3.915273  24145.503978            0.0                 0.0  
min               3.324793    163.000000            0.0                 0.0  
25%              12.390005   1434.000000            0.0                 0.0  
50%              13.106330   7504.000000            0.0                 0.0  
75%              14.614405  10844.500000            0.0                 0.0  
max              18.408230  97802.000000            0.0                 0.0

In [11]:
#users
user_graph =px.bar(google_Ads_Report,x='Session Google Ads campaign',y='Users',
                    text_auto = True,height=500 )
user_graph.show()


#  App installed for may  Google ads campaign have the highest users count.

In [12]:
#  The total number of sessions generated by the Google Ads campaign and The number of sessions where users were actively engaged

bar1 = go.Bar(x=google_Ads_Report['Session Google Ads campaign'],
              y=google_Ads_Report["Sessions"],
              text=google_Ads_Report["Sessions"],
              textposition="auto",
              name="Sessions",
              marker=dict(color='black'))

bar2 = go.Bar(x=google_Ads_Report['Session Google Ads campaign'],
              y=google_Ads_Report["Engaged sessions"],
              text=google_Ads_Report["Engaged sessions"],
              textposition="auto",
              name="Engaged sessions")

layout = go.Layout(title="Number of session generated and actively engaged google Ads campaign", height=800)

fig=go.Figure(data=[bar1,bar2],layout=layout)

fig.show()

In [13]:
# Cost invested in google ads against return as a event count
line1 = go.Scatter(x=google_Ads_Report["Session Google Ads campaign"],
                   y=google_Ads_Report["Google Ads cost"],
                   mode="lines+markers",
                   text=google_Ads_Report["Google Ads cost"],
                   marker=dict(color="teal"),
                   name="Google Ads cost")

line2 = go.Scatter(x=google_Ads_Report["Session Google Ads campaign"],
                   y=google_Ads_Report["Conversions"],
                   mode="lines + markers",
                   text=google_Ads_Report["Conversions"],
                   marker=dict(color="salmon"),
                   name="Conversions")

layout = go.Layout(title="Cost invested in google add againsts conversion", height=600,
                   xaxis=dict(title='sessions google ads campaign'),
                   yaxis=dict(title='Amount invested and number of conversion'))
fig=go.Figure(data=[line1,line2],layout=layout)

fig.show()

**User By Age**

In [14]:
user_by_age

Age  Users  New users  Engaged sessions  Engagement rate  \
0  unknown  14303      13636             24976         0.569098   
1    18-24   4282       3678              7291         0.695308   
2    25-34   2920       2161              3749         0.504780   
3      65+   1422       1081              1640         0.539829   
4    55-64   1403        979              1552         0.519411   
5    35-44   1202        785              1420         0.510424   
6    45-54    810        552               881         0.561862   

   Engaged sessions per user  Average engagement time  Event count  \
0                   1.746207                422.22330       817501   
1                   1.702709                251.16300       309328   
2                   1.283904                 97.24144        90074   
3                   1.153305                 52.30661        24780   
4                   1.106201                 55.37063        25169   
5                   1.181364                 96.08236        33016   
6                   1.087654                 84.54321        18661   

   Conversions  Total revenue  
0        99310              0  
1        53661              0  
2        20172              0  
3         4891              0  
4         4823              0  
5         8111              0  
6         2946              0

In [15]:
user_by_age.dtypes

Age                           object
Users                          int64
New users                      int64
Engaged sessions               int64
Engagement rate              float64
Engaged sessions per user    float64
Average engagement time      float64
Event count                    int64
Conversions                    int64
Total revenue                  int64
dtype: object

In [16]:
user_by_age.sum()

Age                          unknown18-2425-3465+55-6435-4445-54
Users                                                      26342
New users                                                  22872
Engaged sessions                                           41509
Engagement rate                                         3.900712
Engaged sessions per user                               9.261344
Average engagement time                               1058.93055
Event count                                              1318529
Conversions                                               193914
Total revenue                                                  0
dtype: object

In [19]:
user_by_age.describe()

Users     New users  Engaged sessions  Engagement rate  \
count      7.000000      7.000000          7.000000         7.000000   
mean    3763.142857   3267.428571       5929.857143         0.557245   
std     4802.784277   4698.464389       8686.797289         0.065705   
min      810.000000    552.000000        881.000000         0.504780   
25%     1302.500000    882.000000       1486.000000         0.514917   
50%     1422.000000   1081.000000       1640.000000         0.539829   
75%     3601.000000   2919.500000       5520.000000         0.565480   
max    14303.000000  13636.000000      24976.000000         0.695308   

       Engaged sessions per user  Average engagement time    Event count  \
count                   7.000000                 7.000000       7.000000   
mean                    1.323049               151.275793  188361.285714   
std                     0.281679               137.114289  296206.859168   
min                     1.087654                52.306610   18661.000000   
25%                     1.129753                69.956920   24974.500000   
50%                     1.181364                96.082360   33016.000000   
75%                     1.493307               174.202220  199701.000000   
max                     1.746207               422.223300  817501.000000   

        Conversions  Total revenue  
count      7.000000            7.0  
mean   27702.000000            0.0  
std    36282.992857            0.0  
min     2946.000000            0.0  
25%     4857.000000            0.0  
50%     8111.000000            0.0  
75%    36916.500000            0.0  
max    99310.000000            0.0

In [20]:
# Age wise users and new users generated

bar1 = go.Bar(x=user_by_age['Age'],
              y=user_by_age["Users"],
              text=user_by_age["Users"],
              textposition="auto",
              name="Users",
              marker=dict(color='teal'))

bar2 = go.Bar(x=user_by_age['Age'],
              y=user_by_age["New users"],
              text=user_by_age["New users"],
              textposition="auto",
              name="New Users",
              marker=dict(color='salmon'))


layout = go.Layout(title="Age wise users and new users", height=600,
                   xaxis=dict(title='Age Catergory'),
                   yaxis=dict(title='Number of unsers and new users'))
fig=go.Figure(data=[bar1,bar2],layout=layout)

fig.show()



*   Based on the available data, the "unknown" category appears to have the highest number of users (14,303) and new users (13,636). To better understand and classify this category, further investigation and data analysis are needed. Apart from the "unknown" category, there is a notable introduction of users and new users from the 18-20 age group. This demographic represents a significant potential audience for the platform.
*   To capitalize on this opportunity and improve user acquisition from this age group, consider the following suggestions: User survey. Target Marketing. user onboarding. social sharing.



In [21]:
# Sessions engaged and events counts
engaged_sessions=go.Bar(x=user_by_age["Age"],
             y=user_by_age["Engaged sessions"],
              text=user_by_age["Engaged sessions"],
             textposition="auto",
             name="Engaged Sessions",
             marker=dict(color='teal'))

event_count=go.Bar(x=user_by_age["Age"],
             y=user_by_age["Event count"],
                   text=user_by_age["Event count"],
                   textposition="auto",
             name="Event counts",
             marker=dict(color='salmon'))


layout = go.Layout(title="Age group by sessions organished and event happens ", height=600,
                   xaxis=dict(title='Age Catergory'),
                   yaxis=dict(title='Number of sessions and event happens'))
fig=go.Figure(data=[engaged_sessions,event_count],layout=layout)

fig.show()

**User By Language**

In [22]:
user_by_language

Language  Users  New users  Engaged sessions  Engagement rate  \
0     English  22495      21990             40639         0.595147   
1       Hindi    586        552               798         0.406314   
2     Marathi     85         84                98         0.426087   
3    Gujarati     78         77               100         0.448430   
4      Telugu     43         42                56         0.455285   
5       Tamil     36         36                43         0.518072   
6   Malayalam     17         15                36         0.654545   
7     Bengali     14         11                18         0.600000   
8     Chinese     13         13                13         1.000000   
9     Kannada     13         12                31         0.500000   
10    Panjabi      9          9                17         0.708333   
11    Persian      8          8                 6         0.400000   
12    Spanish      6          6                 8         0.470588   
13    Finnish      4          3                 4         0.571429   
14   Japanese      4          4                 3         0.428571   
15      Oriya      4          4                 2         0.666667   
16  Afrikaans      1          1                 1         1.000000   
17   Assamese      1          0                 1         1.000000   
18     German      1          1                 0         0.000000   
19      Malay      1          1                 1         1.000000   
20     Nepali      1          1                 1         1.000000   
21    Russian      1          1                 0         0.000000   
22       Urdu      1          1                 0         0.000000   
23   Sanskrit      0          0                 0         0.000000   

    Engaged sessions per user  Average engagement time  Event count  \
0                    1.806579                341.36350      1297970   
1                    1.361775                 60.03413        13523   
2                    1.152941                 38.48235         1589   
3                    1.282051                 46.53846         1794   
4                    1.302326                 36.65116          812   
5                    1.194444                 45.86111          615   
6                    2.117647                161.94120          548   
7                    1.285714                 50.07143          217   
8                    1.000000                136.76920          138   
9                    2.384615                249.07690          680   
10                   1.888889                 92.44444          229   
11                   0.750000                 28.25000           99   
12                   1.333333                 22.16667          113   
13                   1.000000                 89.25000           64   
14                   0.750000                  9.25000           49   
15                   0.500000                  7.50000           29   
16                   1.000000                 37.00000           12   
17                   1.000000                 42.00000            6   
18                   0.000000                  0.00000            5   
19                   1.000000                  2.00000            7   
20                   1.000000                  5.00000            7   
21                   0.000000                 70.00000           12   
22                   0.000000                  1.00000            7   
23                   0.000000                  0.00000            4   

    Conversions  Total revenue  
0        189946              0  
1          2699              0  
2           323              0  
3           327              0  
4           170              0  
5           115              0  
6            71              0  
7            39              0  
8            20              0  
9            75              0  
10           35              0  
11           23              0  
12           22              0  
13           11 

In [23]:
user_by_language.dtypes

Language                      object
Users                          int64
New users                      int64
Engaged sessions               int64
Engagement rate              float64
Engaged sessions per user    float64
Average engagement time      float64
Event count                    int64
Conversions                    int64
Total revenue                  int64
dtype: object

In [24]:
user_by_language.isnull().sum()

Language                     0
Users                        0
New users                    0
Engaged sessions             0
Engagement rate              0
Engaged sessions per user    0
Average engagement time      0
Event count                  0
Conversions                  0
Total revenue                0
dtype: int64

In [25]:
user_by_language.sum()

Language                     EnglishHindiMarathiGujaratiTeluguTamilMalayala...
Users                                                                    23422
New users                                                                22872
Engaged sessions                                                         41876
Engagement rate                                                      12.849468
Engaged sessions per user                                            25.110314
Average engagement time                                             1572.65055
Event count                                                            1318529
Conversions                                                             193914
Total revenue                                                                0
dtype: object

In [26]:
user_by_language.describe()

Users     New users  Engaged sessions  Engagement rate  \
count     24.000000     24.000000         24.000000        24.000000   
mean     975.916667    953.000000       1744.833333         0.535394   
std     4585.079838   4482.258359       8285.991965         0.321298   
min        0.000000      0.000000          0.000000         0.000000   
25%        1.000000      1.000000          1.000000         0.421144   
50%        7.000000      7.000000          7.000000         0.509036   
75%       21.750000     20.250000         37.750000         0.677083   
max    22495.000000  21990.000000      40639.000000         1.000000   

       Engaged sessions per user  Average engagement time   Event count  \
count                  24.000000                24.000000  2.400000e+01   
mean                    1.046263                65.527106  5.493871e+04   
std                     0.642224                83.316618  2.647787e+05   
min                     0.000000                 0.000000  4.000000e+00   
25%                     0.750000                 8.812500  1.075000e+01   
50%                     1.000000                40.241175  1.060000e+02   
75%                     1.310078                74.812500  6.312500e+02   
max                     2.384615               341.363500  1.297970e+06   

         Conversions  Total revenue  
count      24.000000           24.0  
mean     8079.750000            0.0  
std     38741.203482            0.0  
min         1.000000            0.0  
25%         2.000000            0.0  
50%        21.000000            0.0  
75%        85.000000            0.0  
max    189946.000000            0.0

In [27]:
# Top 10 Language prefrence
user_by_language_10 = user_by_language.sort_values(by=["Users","New users"],ascending =False)

top_10_language=user_by_language_10.head(10)

bar1=go.Bar(x=top_10_language["Language"],
             y=top_10_language["Users"],
              text=top_10_language["Users"],
             textposition="auto",
             name="Users",
             marker=dict(color='teal'))
bar2=go.Bar(x=top_10_language["Language"],
             y=top_10_language["New users"],
                   text=top_10_language["New users"],
                   textposition="auto",
             name="New Users",
             marker=dict(color='salmon'))


layout = go.Layout(title="Top 10 language prefrences by users and new users ", height=600,
                   xaxis=dict(title='Languages'),
                   yaxis=dict(title='Number of users / new users'))
fig=go.Figure(data=[bar1,bar2],layout=layout)

fig.show()

# Top prefrences language was english

In [28]:
user_by_language_10.head(10)

Language  Users  New users  Engaged sessions  Engagement rate  \
0    English  22495      21990             40639         0.595147   
1      Hindi    586        552               798         0.406314   
2    Marathi     85         84                98         0.426087   
3   Gujarati     78         77               100         0.448430   
4     Telugu     43         42                56         0.455285   
5      Tamil     36         36                43         0.518072   
6  Malayalam     17         15                36         0.654545   
7    Bengali     14         11                18         0.600000   
8    Chinese     13         13                13         1.000000   
9    Kannada     13         12                31         0.500000   

   Engaged sessions per user  Average engagement time  Event count  \
0                   1.806579                341.36350      1297970   
1                   1.361775                 60.03413        13523   
2                   1.152941                 38.48235         1589   
3                   1.282051                 46.53846         1794   
4                   1.302326                 36.65116          812   
5                   1.194444                 45.86111          615   
6                   2.117647                161.94120          548   
7                   1.285714                 50.07143          217   
8                   1.000000                136.76920          138   
9                   2.384615                249.07690          680   

   Conversions  Total revenue  
0       189946              0  
1         2699              0  
2          323              0  
3          327              0  
4          170              0  
5          115              0  
6           71              0  
7           39              0  
8           20              0  
9           75              0

**User By Interest **

In [29]:
user_by_intrest.head(10)

Interests  Users  New users  \
0                                       Shoppers  10950       9256   
1  Media & Entertainment/Comics & Animation Fans  10946       9247   
2                  Technology/Mobile Enthusiasts  10934       9239   
3              Food & Dining/Cooking Enthusiasts   8410       6970   
4        Sports & Fitness/Health & Fitness Buffs   5844       4580   
5                Lifestyles & Hobbies/Pet Lovers   5785       4768   
6             Media & Entertainment/Music Lovers   5488       4504   
7                Media & Entertainment/TV Lovers   5363       4312   
8              News & Politics/Avid News Readers   5352       4110   
9       Lifestyles & Hobbies/Outdoor Enthusiasts   5247       4271   

   Engaged sessions  Engagement rate  Engaged sessions per user  \
0             15652         0.581534                   1.429406   
1             15680         0.583008                   1.432487   
2             15619         0.582451                   1.428480   
3             12332         0.602325                   1.466350   
4              8226         0.588328                   1.407598   
5              7786         0.592542                   1.345895   
6              8011         0.659016                   1.459730   
7              7423         0.610444                   1.384113   
8              7079         0.552184                   1.322683   
9              7441         0.609369                   1.418144   

   Average engagement time  Event count  Conversions  Total revenue  
0                 162.8347       490664        86846              0  
1                 165.1772       491025        86845              0  
2                 162.6945       489353        86742              0  
3                 176.9567       409713        73814              0  
4                 155.1451       257831        43074              0  
5                 124.0411       236538        34111              0  
6                 175.9355       282690        48962              0  
7                 163.5937       250592        38664              0  
8                 139.2730       176306        34901              0  
9                 126.8186       222219        29529              0

In [30]:
user_by_intrest.dtypes

Interests                     object
Users                          int64
New users                      int64
Engaged sessions               int64
Engagement rate              float64
Engaged sessions per user    float64
Average engagement time      float64
Event count                    int64
Conversions                    int64
Total revenue                  int64
dtype: object

In [31]:
user_by_intrest.isnull().sum()

Interests                    0
Users                        0
New users                    0
Engaged sessions             0
Engagement rate              0
Engaged sessions per user    0
Average engagement time      0
Event count                  0
Conversions                  0
Total revenue                0
dtype: int64

In [32]:
user_by_intrest.sum()

Interests                    ShoppersMedia & Entertainment/Comics & Animati...
Users                                                                   214583
New users                                                               170104
Engaged sessions                                                        300280
Engagement rate                                                      55.107979
Engaged sessions per user                                           119.186295
Average engagement time                                            19538.33655
Event count                                                            9840303
Conversions                                                            1736648
Total revenue                                                                0
dtype: object

In [33]:
# Top 10 User intrest
user_by_intrest_sort=user_by_intrest.sort_values(by=["Users","New users"],ascending=False)
top10_userby_intrest=user_by_intrest_sort.head(10)
top10_userby_intrest

bar1=go.Bar(x=top10_userby_intrest["Interests"],
            y=top10_userby_intrest["Users"],
            text=top10_userby_intrest["Users"],
            textposition="auto",
            name="Users",
            marker=dict(color="teal"))


bar2=go.Bar(x=top10_userby_intrest["Interests"],
            y=top10_userby_intrest["New users"],
            text=top10_userby_intrest["New users"],
            textposition="auto",
            name="New users",
            marker=dict(color="salmon"))

layout = go.Layout(title="Top 10 Intrest users and user users attract ", height=600,
                   xaxis=dict(title='Intrest'),
                   yaxis=dict(title='Number of users / new users'))
fig=go.Figure(data=[bar1,bar2],layout=layout)

fig.show()

In [34]:
# Top 10 intrest with the highest engaged sessions

line1 = go.Scatter(x=top10_userby_intrest["Interests"],
                   y=top10_userby_intrest["Engaged sessions"],
                   mode="lines+markers",  # Show both lines and markers
                   text=top10_userby_intrest["Engaged sessions"],
                   textposition="top center",  # Position of the data labels
                   name="Engaged sessions",
                   marker=dict(color="teal"))

layout = go.Layout(title="Top 10 Interests with Highest Engaged Sessions", height=600,
                   xaxis=dict(title='Interest', showline=True, showticklabels=True),
                   yaxis=dict(title='Number of Engaged Sessions', showline=True, showticklabels=True))

fig = go.Figure(data=[line1], layout=layout)

fig.show()

In [35]:
# Top 10 intrest with the highest conversions

line1 = go.Scatter(x=top10_userby_intrest["Interests"],
                   y=top10_userby_intrest["Conversions"],
                   mode="lines+markers",  # Show both lines and markers
                   text=top10_userby_intrest["Conversions"],
                   textposition="top center",  # Position of the data labels
                   name="Conversions",
                   marker=dict(color="salmon"))

layout = go.Layout(title="Top 10 Interests with Highest conversions ", height=600,
                   xaxis=dict(title='Interest', showline=True, showticklabels=True),
                   yaxis=dict(title='Number of conversions', showline=True, showticklabels=True))

fig = go.Figure(data=[line1], layout=layout)

fig.show()

**Gender Report**

In [36]:
gender_report

Gender  Users  New users  Engaged sessions  Engagement rate  \
0  unknown  13142      12691             23161         0.564077   
1     male   7218       5877             10467         0.543091   
2   female   4944       4304              7877         0.637710   

   Engaged sessions per user  Average engagement time  Event count  \
0                   1.762365                 439.5776       761771   
1                   1.450125                 128.2319       282504   
2                   1.593244                 208.7407       274254   

   Conversions  Total revenue  
0        93180              0  
1        65651              0  
2        35083              0

In [37]:
gender_report.dtypes

Gender                        object
Users                          int64
New users                      int64
Engaged sessions               int64
Engagement rate              float64
Engaged sessions per user    float64
Average engagement time      float64
Event count                    int64
Conversions                    int64
Total revenue                  int64
dtype: object

In [38]:
gender_report.isnull().sum()

Gender                       0
Users                        0
New users                    0
Engaged sessions             0
Engagement rate              0
Engaged sessions per user    0
Average engagement time      0
Event count                  0
Conversions                  0
Total revenue                0
dtype: int64

In [39]:
#Gender wise user percentage


pie_chart = go.Pie(labels=gender_report["Gender"],
                   values=gender_report["Users"],
                   marker=dict(colors=["teal","salmon","Olive"]),
                   textinfo='label+percent',
                   textposition='inside',
                   hoverinfo='label+percent',
                   hole=0.4)
layout = go.Layout(title='User gender wise percentage ')

fig = go.Figure(data=[pie_chart], layout=layout)
fig.show()

In [40]:
# New users gender wise percentage


pie_chart = go.Pie(labels=gender_report["Gender"],
                   values=gender_report["New users"],
                   marker=dict(colors=["teal","salmon","Olive"]),
                   textinfo='label+percent',
                   textposition='outside',
                   hoverinfo='label+percent',
                   hole=0.4)
layout = go.Layout(title='New user gender')

fig = go.Figure(data=[pie_chart], layout=layout)
fig.show()

In [41]:
# average time spend by gender
bar1=go.Bar(x=gender_report["Gender"],
            y=gender_report["Average engagement time"]/60,
            text=gender_report["Average engagement time"]/60,
            textposition="auto",
            name="Avergae engament time in mins",
            marker=dict(color="Olive")
            )

layout = go.Layout(title="Average time watched in mins ", height=600,
                   xaxis=dict(title='Gender'),
                   yaxis=dict(title='Average time in min'))
fig=go.Figure(data=[bar1],layout=layout)

fig.show()

**Citiwwise Report**

In [42]:
citiwise_report

Town/City  Users  New users  Engaged sessions  Engagement rate  \
0     Bengaluru   6097       5685             15013         0.769385   
1         Patna   1594       1467              2127         0.440646   
2     Hyderabad   1038        920              1578         0.569264   
3        Indore    983        915              1241         0.426460   
4       Lucknow    897        839              1125         0.450180   
..          ...    ...        ...               ...              ...   
569     Titwala      0          0                 0         0.000000   
570     Vagator      0          0                 0         0.000000   
571   Vellakoil      0          0                 0         0.000000   
572      Wardha      0          0                 0         0.000000   
573  Washington      0          0                 0         0.000000   

     Engaged sessions per user  Average engagement time  Event count  \
0                     2.462359                762.20550       607200   
1                     1.334379                 98.22208        38830   
2                     1.520231                243.69080        96826   
3                     1.262462                 67.89115        21383   
4                     1.254181                 83.40580        21041   
..                         ...                      ...          ...   
569                   0.000000                  0.00000            1   
570                   0.000000                  0.00000           16   
571                   0.000000                  0.00000            1   
572                   0.000000                  0.00000            1   
573                   0.000000                  0.00000            1   

     Conversions  Total revenue  
0          62939              0  
1           6980              0  
2          34103              0  
3           4121              0  
4           3650              0  
..           ...            ...  
569            1              0  
570           16              0  
571            0              0  
572            1              0  
573            1              0  

[574 rows x 10 columns]

In [43]:
citiwise_report.dtypes

Town/City                     object
Users                          int64
New users                      int64
Engaged sessions               int64
Engagement rate              float64
Engaged sessions per user    float64
Average engagement time      float64
Event count                    int64
Conversions                    int64
Total revenue                  int64
dtype: object

In [44]:
citiwise_report.isnull().sum()

Town/City                    0
Users                        0
New users                    0
Engaged sessions             0
Engagement rate              0
Engaged sessions per user    0
Average engagement time      0
Event count                  0
Conversions                  0
Total revenue                0
dtype: int64

In [45]:
citiwise_report.describe()

Users    New users  Engaged sessions  Engagement rate  \
count   574.000000   574.000000         574.00000       574.000000   
mean     46.203833    39.846690          71.61324         0.489662   
std     290.091708   267.115856         656.03377         0.341660   
min       0.000000     0.000000           0.00000         0.000000   
25%       1.000000     1.000000           1.00000         0.250000   
50%       2.000000     1.000000           2.00000         0.497547   
75%       7.750000     5.000000           7.00000         0.724026   
max    6097.000000  5685.000000       15013.00000         1.000000   

       Engaged sessions per user  Average engagement time    Event count  \
count                 574.000000               574.000000     574.000000   
mean                    0.884915                94.093768    2297.088850   
std                     0.750856               202.850678   26179.153773   
min                     0.000000                 0.000000       1.000000   
25%                     0.500000                 7.000000      10.000000   
50%                     1.000000                27.500000      35.000000   
75%                     1.000000                98.402298     162.750000   
max                     7.000000              2721.000000  607200.000000   

        Conversions  Total revenue  
count    574.000000          574.0  
mean     337.829268            0.0  
std     3074.301547            0.0  
min        0.000000            0.0  
25%        2.000000            0.0  
50%        5.000000            0.0  
75%       23.000000            0.0  
max    62939.000000            0.0

In [46]:
#Top 10 cities attract users

citiwise_report_sort=citiwise_report.sort_values(by=["Users","New users"],ascending =False)
top10_citi = citiwise_report_sort.head(10)
top10_citi

bar1=go.Bar(x=top10_citi["Town/City"],
            y=top10_citi["Users"],
            text=top10_citi["Users"],
            textposition="auto",
            name="Users",
            marker=dict(color="teal"))

bar2=go.Bar(x=top10_citi["Town/City"],
            y=top10_citi["New users"],
            text=top10_citi["New users"],
            textposition="auto",
            name="New users",
            marker=dict(color="Olive"))

layout = go.Layout(title="Top 10 cities attarct users", height=600,
                   xaxis=dict(title='Town/city'),
                   yaxis=dict(title='Number of users / new users'))
fig=go.Figure(data=[bar1,bar2],layout=layout)

fig.show()

In [47]:
#Top 10 cities by engaged session impact of Event count and conversion

citiwise_report_sort=citiwise_report.sort_values(by=["Engaged sessions"],ascending =False)
top10_citi = citiwise_report_sort.head(10)
top10_citi

bar1=go.Bar(x=top10_citi["Town/City"],
            y=top10_citi["Engaged sessions"],
            text=top10_citi["Engaged sessions"],
            textposition="auto",
            name="Engaged sessions",
            marker=dict(color="teal"))

bar2=go.Bar(x=top10_citi["Town/City"],
            y=top10_citi["Event count"],
            text=top10_citi["Event count"],
            textposition="auto",
            name="Event count",
            marker=dict(color="Olive"))

bar3=go.Bar(x=top10_citi["Town/City"],
            y=top10_citi["Conversions"],
            text=top10_citi["Conversions"],
            textposition="auto",
            name="Conversions",
            marker=dict(color="salmon"))
layout = go.Layout(title="Top 10 cities by engaged session impact of Event count and conversion", height=600,
                   xaxis=dict(title='Town/city'),
                   yaxis=dict(title='Numbers count'))
fig=go.Figure(data=[bar1,bar2,bar3],layout=layout)

fig.show()

**Demographic Report**

In [48]:
demographics_report.head()

Country  Users  New users  Engaged sessions  Engagement rate  \
0           India  23024      22528             41479         0.593626   
1   United States    272        213               197         0.491272   
2          Canada     37         18                25         0.416667   
3       (not set)     36         36                17         0.459459   
4  United Kingdom     20          8                13         0.371429   

   Engaged sessions per user  Average engagement time  Event count  \
0                   1.801555                334.81660      1312097   
1                   0.724265                 50.96324         3157   
2                   0.675676                 43.21622          410   
3                   0.472222                 24.80556          241   
4                   0.650000                 61.85000          289   

   Conversions  Total revenue  
0       192766              0  
1          643              0  
2          121              0  
3           54              0  
4           43              0

In [49]:
demographics_report.dtypes

Country                       object
Users                          int64
New users                      int64
Engaged sessions               int64
Engagement rate              float64
Engaged sessions per user    float64
Average engagement time      float64
Event count                    int64
Conversions                    int64
Total revenue                  int64
dtype: object

In [50]:
demographics_report.isnull().sum()

Country                      0
Users                        0
New users                    0
Engaged sessions             0
Engagement rate              0
Engaged sessions per user    0
Average engagement time      0
Event count                  0
Conversions                  0
Total revenue                0
dtype: int64

In [51]:
demographics_report.describe()

Users     New users  Engaged sessions  Engagement rate  \
count     47.000000     47.000000         47.000000        47.000000   
mean     500.148936    486.638298        890.340426         0.561662   
std     3357.099816   3285.103773       6049.238031         0.358970   
min        0.000000      0.000000          0.000000         0.000000   
25%        1.000000      0.000000          1.000000         0.364286   
50%        2.000000      1.000000          1.000000         0.533333   
75%        7.000000      3.000000          6.000000         1.000000   
max    23024.000000  22528.000000      41479.000000         1.000000   

       Engaged sessions per user  Average engagement time   Event count  \
count                  47.000000                47.000000  4.700000e+01   
mean                    0.835367                48.376626  2.805381e+04   
std                     0.568228                93.511799  1.913692e+05   
min                     0.000000                 0.000000  1.000000e+00   
25%                     0.550000                 6.800000  7.000000e+00   
50%                     1.000000                17.000000  2.300000e+01   
75%                     1.000000                48.428570  8.650000e+01   
max                     2.666667               536.000000  1.312097e+06   

         Conversions  Total revenue  
count      47.000000           47.0  
mean     4125.829787            0.0  
std     28114.328609            0.0  
min         0.000000            0.0  
25%         2.000000            0.0  
50%         4.000000            0.0  
75%        15.500000            0.0  
max    192766.000000            0.0

In [52]:
demographic_sort=demographics_report.sort_values(by=["Users","New users"],ascending=False)
top10_demographic=demographics_report.head(10)
top10_demographic

bar1=go.Bar(x=top10_demographic["Country"],
            y=top10_demographic["Users"],
            text=top10_demographic["Users"],
            textposition="auto",
            name="Users",
            marker=dict(color="teal"))


bar2=go.Bar(x=top10_demographic["Country"],
            y=top10_demographic["New users"],
            text=top10_demographic["New users"],
            textposition="auto",
            name="New users",
            marker=dict(color="salmon"))

layout = go.Layout(title="Top 10 country who install application ", height=600,
                   xaxis=dict(title='Country'),
                   yaxis=dict(title='Number of users / new users'))
fig=go.Figure(data=[bar1,bar2],layout=layout)

fig.show()

**Traffic Aquisition**

In [53]:
traffic_aquisition

Session default channel group  Users  Sessions  Engaged sessions  \
0                    Unassigned  20263     13448              1481   
1                       Display   9613     18292             10613   
2                Organic Search   7689     21241             17814   
3                        Direct   4042     13220              7649   
4                   Paid Search   2909      6788              3452   
5                Organic Social     11        16                12   

   Average engagement time per session  Engaged sessions per user  \
0                             34.11704                   0.073089   
1                             28.52198                   1.104026   
2                            195.94340                   2.316816   
3                            177.17060                   1.892380   
4                             36.65321                   1.186662   
5                             60.06250                   1.090909   

   Events per session  Engagement rate  Event count  Conversions  \
0           18.023130         0.110128       242375       114161   
1            9.069320         0.580199       165896        20031   
2           29.302290         0.838661       622410        33612   
3           17.135850         0.578593       226536        18496   
4            8.989982         0.508544        61024         7595   
5           18.000000         0.750000          288           19   

   Total revenue  
0              0  
1              0  
2              0  
3              0  
4              0  
5              0

In [54]:
traffic_aquisition.dtypes

Session default channel group           object
Users                                    int64
Sessions                                 int64
Engaged sessions                         int64
Average engagement time per session    float64
Engaged sessions per user              float64
Events per session                     float64
Engagement rate                        float64
Event count                              int64
Conversions                              int64
Total revenue                            int64
dtype: object

In [55]:
user_acquisition.isnull().sum()

First user default channel group    0
New users                           0
Engaged sessions                    0
Engagement rate                     0
Engaged sessions per user           0
Average engagement time             0
Event count                         0
Conversions                         0
Total revenue                       0
dtype: int64

In [56]:
user_acquisition.describe()

New users  Engaged sessions  Engagement rate  \
count     6.000000          6.000000         6.000000   
mean   3812.000000       6860.666667         0.614435   
std    4083.739659       6894.604867         0.201367   
min      10.000000         13.000000         0.318808   
25%     719.500000       2316.250000         0.491827   
50%    2464.000000       4691.500000         0.633339   
75%    6495.250000      10249.750000         0.790425   
max    9957.000000      18141.000000         0.813680   

       Engaged sessions per user  Average engagement time    Event count  \
count                   6.000000                 6.000000       6.000000   
mean                    2.262367               461.322532  219754.833333   
std                     1.420806               437.111008  284869.104021   
min                     1.206107                58.862090     248.000000   
25%                     1.339538               113.003350   45489.250000   
50%                     1.859759               339.806400  143408.500000   
75%                     2.340622               732.334325  221780.500000   
max                     4.981538              1128.881000  770710.000000   

         Conversions  Total revenue  
count       6.000000            6.0  
mean    32319.000000            0.0  
std     40929.227185            0.0  
min        27.000000            0.0  
25%      4284.250000            0.0  
50%     22931.500000            0.0  
75%     35848.750000            0.0  
max    109801.000000            0.0

In [57]:
# New User Acquisition
new_user_acq = go.Pie(labels=user_acquisition["First user default channel group"],
                     values=user_acquisition["New users"],
                     marker=dict(colors=["teal","salmon","Olive","DarkSeaGreen","RosyBrown","DarkSlateGray"]),
                     textinfo='label+percent',
                     textposition='outside',
                     hoverinfo='label+percent',
                     hole=0.4)
layout = go.Layout(title='New User Acquisition')

fig = go.Figure(data=[new_user_acq], layout=layout)
fig.show()

**Page And Report**

In [58]:
pages_and_screens_report.head()

Page path and screen class   Views  Users  Views per user  \
0                    Flutter  156708   8726       17.958740   
1               MainActivity   44326   8978        4.937180   
2                      feeds   18514   4358        4.248279   
3                      login   16883   7291        2.315595   
4          my_rewards_screen   15381   2045        7.521271   

   Average engagement time  Event count  Conversions  Total revenue  
0                 83.41222       203901          328              0  
1                 78.29216        53374          101              0  
2                 61.60005        37628          253              0  
3                 34.88177        40772          435              0  
4                 94.17995        32910            5              0

In [59]:
pages_and_screens_report.dtypes

Page path and screen class     object
Views                           int64
Users                           int64
Views per user                float64
Average engagement time       float64
Event count                     int64
Conversions                     int64
Total revenue                   int64
dtype: object

In [60]:
pages_and_screens_report.isnull().sum()

Page path and screen class    0
Views                         0
Users                         0
Views per user                0
Average engagement time       0
Event count                   0
Conversions                   0
Total revenue                 0
dtype: int64

In [61]:
pages_and_screens_report.describe()

Views        Users  Views per user  Average engagement time  \
count      42.000000    42.000000       42.000000                42.000000   
mean     7081.261905  1524.761905        3.036123                48.898515   
std     24926.556037  2628.069168        3.001363                58.750556   
min         0.000000     2.000000        0.000000                 0.001093   
25%        69.250000    23.250000        1.510656                 9.340753   
50%       736.500000   301.000000        1.864079                35.592545   
75%      2734.750000  1402.500000        3.696037                67.387118   
max    156708.000000  9145.000000       17.958740               304.333300   

         Event count   Conversions  Total revenue  
count      42.000000     42.000000           42.0  
mean    13417.547619   2145.690476            0.0  
std     36182.217422  13652.911458            0.0  
min         4.000000      0.000000            0.0  
25%       146.750000      0.000000            0.0  
50%      1330.500000      1.500000            0.0  
75%      6303.750000     17.750000            0.0  
max    203901.000000  88518.000000            0.0

In [62]:
 #Top 10 pages and screens report
pages_and_screens_report_sort = pages_and_screens_report.sort_values(by="Views",ascending=False)
top10_pages_and_screens=pages_and_screens_report_sort.head(10)
top10_pages_and_screens

bar1=go.Bar(x=top10_pages_and_screens["Page path and screen class"],
            y=top10_pages_and_screens["Views"],
            text=top10_pages_and_screens["Views"],
            textposition="auto",
            name="Views",
            marker=dict(color="teal"))

layout = go.Layout(title="Top 10 pages and screens by views", height=600,
                   xaxis=dict(title='pages path and screen'),
                   yaxis=dict(title='Number of views'))
fig=go.Figure(data=[bar1],layout=layout)

fig.show()

In [63]:
#Below 10 pages and screens

pages_and_screens_report_sort = pages_and_screens_report.sort_values(by="Views",ascending=True)
below10_pages_and_screens=pages_and_screens_report_sort.head(10)
below10_pages_and_screens

bar1=go.Bar(x=below10_pages_and_screens["Page path and screen class"],
            y=below10_pages_and_screens["Views"],
            text=below10_pages_and_screens["Views"],
            textposition="auto",
            name="Views",
            marker=dict(color="teal"))

layout = go.Layout(title="below 10 pages and screens by views", height=600,
                   xaxis=dict(title='pages path and screen'),
                   yaxis=dict(title='Number of views'))
fig=go.Figure(data=[bar1],layout=layout)

fig.show()